lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -155,223 +182,188 @@
-        } else {
-            return resolvedLocale;
+        // our best fit matcher for WinGlob will return the default locale if given an unsupported locale
+        if (isPlatformUsingWinGlob) {
+            const defaultLocale = GetDefaultLocale();
+            if (defaultLocale === locale) {
+                return resolvedLocale;
+            } else if (defaultLocale === resolvedLocale) {
+                return undefined;
+            } else {
+                return resolvedLocale;
+            }
         }
+
+        return resolvedLocale;
     }
 
-    var resolveLocaleLookup = function (localeWithoutSubtags) {
-        let resolvedLocale = platform.localeLookupCache.get(localeWithoutSubtags);
+    const __resolveLocaleLookup = new IntlCache();
+    const resolveLocaleLookup = function (locale) {
+        let resolvedLocale = __resolveLocaleLookup.get(locale);
         if (resolvedLocale === undefined) {
-            resolvedLocale = platform.resolveLocaleLookup(localeWithoutSubtags);
+            // the lookup matcher in WinGlob uses ResolveLocaleName, an NLS function
+            const nlsLocale = toNLS(locale);
+            resolvedLocale = platform.resolveLocaleLookup(locale);
             if (resolvedLocale === null) {
-                if (!LANG_TAG_BASE_RE) {
-                    InitializeLangTagREs();
-                }
-                let match = platform.builtInRegexMatch(localeWithoutSubtags, LANG_TAG_BASE_RE);
-                // match: [1] language; [2] script; [3] region (e.g. en-Latn-US)
-                resolvedLocale = match[1]
-                    + (match[2] ? ('-' + match[2]) : '')
-                    + (match[3] ? ('-' + match[3]) : '');
+                resolvedLocale = platform.builtInRegexMatch(locale, LANG_TAG_BASE_RE)[0];
+            } else {
+                resolvedLocale = platform.builtInRegexMatch(fromNLS(resolvedLocale), LANG_TAG_BASE_RE)[0];
             }
             // If resolvedLocale is undefined, cache that we got undefined
             // so we don't try to resolve for `locale` in future.
-            platform.localeLookupCache.set(localeWithoutSubtags, resolvedLocale === undefined ? NOT_FOUND : resolvedLocale);
-        } else if (resolvedLocale === NOT_FOUND) {
+            __resolveLocaleLookup.set(locale, resolvedLocale === undefined ? LANGUAGE_NOT_FOUND : resolvedLocale);
+        } else if (resolvedLocale === LANGUAGE_NOT_FOUND) {
             resolvedLocale = undefined;
         }
+
         return resolvedLocale;
     }
 
-    var getExtensionSubtags = function (locale) {
-        if (!LANG_TAG_EXT_RE) {
-            InitializeLangTagREs();
+    const UnicodeExtensionSubtags = function (extensionString) {
+        if (!extensionString) {
+            return [];
         }
 
-        const match = platform.builtInRegexMatch(locale, LANG_TAG_EXT_RE);
-        if (!match) {
-            return undefined;
+        const extensions = _.filter(_.split(extensionString, "-"), (el) => !!el);
```

Comment:
Just a nitpick but you don't need parens around `el` in the lambda parameter list.

Reply:
Personally, I have always used parens around single-argument lambdas just because I think its nicer to match no-argument and multiple-argument ones. If the general current coding style for new JS in ChakraCore says to prefer un-parenthesized single-argument lists, then I can change it.

Reply:
There is no style guide as far as I know. This is your call. 👍

Reply:
lol yeah there's no style guide at all for JS. It generally seems to be same rules as for C++ (where applicable) but we're okay with open braces on the end of a line rather than on a line by itself, which is helpful because JS uses a lot of open braces.

Reply:
FWIW I agree that keeping the parens for single-argument lists is more consistent and thus preferable.